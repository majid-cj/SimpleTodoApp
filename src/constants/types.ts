import {strings} from '../i18n';

export type Action = {
  type: string;
  payload: any;
};

export type Task = {
  id?: string;
  title: string;
  date?: string;
};

export type Todo = {
  id?: string;
  task: string;
  description: string;
  done?: boolean;
};

export type TaskList = Array<Task>;

export type TodoList = Array<Todo>;

export type TaskResponse = {
  data: Task;
};

export type TodoResponse = {
  data: Todo;
};

export type TaskListResponse = {
  data: TaskList;
};

export type TodoListResponse = {
  data: TodoList;
};

export type Error = {
  message: string;
};

export class ErrorWrapper implements Error {
  message: string = strings('errors.general');

  constructor(code: string = 'general') {
    this.message = strings(`errors.${code}`);
  }
}
