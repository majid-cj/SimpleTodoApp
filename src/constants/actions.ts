// Tasks Actions
export const GET_ALL_TASKS_REQUEST = 'GET_ALL_TASKS_REQUEST';
export const GET_ALL_TASKS_SUCCESS = 'GET_ALL_TASKS_SUCCESS';
export const GET_ALL_TASKS_FAILURE = 'GET_ALL_TASKS_FAILURE';
export const GET_ALL_TASKS_CLEAR_STATE = 'GET_ALL_TASKS_CLEAR_STATE';

export const NEW_TASK_REQUEST = 'NEW_TASK_REQUEST';
export const NEW_TASK_SUCCESS = 'NEW_TASK_SUCCESS';
export const NEW_TASK_FAILURE = 'NEW_TASK_FAILURE';
export const NEW_TASK_CLEAR_STATE = 'NEW_TASK_CLEAR_STATE';

// Todo Actions
export const GET_ALL_TODO_REQUEST = 'GET_ALL_TODO_REQUEST';
export const GET_ALL_TODO_SUCCESS = 'GET_ALL_TODO_SUCCESS';
export const GET_ALL_TODO_FAILURE = 'GET_ALL_TODO_FAILURE';
export const GET_ALL_TODO_CLEAR_STATE = 'GET_ALL_TODO_CLEAR_STATE';

export const NEW_TODO_REQUEST = 'NEW_TODO_REQUEST';
export const NEW_TODO_SUCCESS = 'NEW_TODO_SUCCESS';
export const NEW_TODO_FAILURE = 'NEW_TODO_FAILURE';
export const NEW_TODO_CLEAR_STATE = 'NEW_TODO_CLEAR_STATE';

export const UPDATE_TODO_REQUEST = 'UPDATE_TODO_REQUEST';
export const UPDATE_TODO_SUCCESS = 'UPDATE_TODO_SUCCESS';
export const UPDATE_TODO_FAILURE = 'UPDATE_TODO_FAILURE';
export const UPDATE_TODO_CLEAR_STATE = 'UPDATE_TODO_CLEAR_STATE';

// Theme Actions
export const UPDATE_APP_THEME = 'UPDATE_APP_THEME';
