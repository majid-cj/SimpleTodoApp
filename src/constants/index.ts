export * from './actions';
export * from './types';
export * from './urls';
