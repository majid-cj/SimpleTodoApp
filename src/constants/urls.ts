import {AxiosRequestConfig} from 'axios';

const APP_URL = 'http://192.168.30.46:8080/';

export const apiConfig: AxiosRequestConfig = {
  headers: {
    'Content-Type': 'application/json',
  },
};

export const LIST_TASKS_URLS = `${APP_URL}task`;
export const CREATE_TASKS_URLS = `${APP_URL}task`;

export const LIST_TODO_URLS = (task: string) => `${APP_URL}task/todo/${task}`;
export const CREATE_TODO_URLS = `${APP_URL}task/todo`;
export const UPDATE_TODO_URLS = `${APP_URL}task/todo`;
