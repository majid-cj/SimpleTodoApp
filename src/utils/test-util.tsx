import React from 'react';
import {render as AppRender} from '@testing-library/react-native';
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';
import taskListReducer from '../reducers/task-list';
import newTaskReducer from '../reducers/new-task';
import themeReducer from '../reducers/theme';
import todoListReducer from '../reducers/todo-list';
import todoReducer from '../reducers/new-todo';
import updateTodoReducer from '../reducers/update-todo';
import {createStackNavigator} from '@react-navigation/stack';
import {NavigationContainer} from '@react-navigation/native';

function render(
  ui: any,
  {
    preloadedState,
    store = configureStore({
      reducer: {
        taskList: taskListReducer,
        task: newTaskReducer,
        theme: themeReducer,
        todoList: todoListReducer,
        todo: todoReducer,
        updateTodo: updateTodoReducer,
      },
      preloadedState,
    }),
  } = {},
) {
  function Wrapper({children}: {children: any}) {
    return <Provider store={store}>{children}</Provider>;
  }
  return AppRender(ui, {wrapper: Wrapper});
}

export * from '@testing-library/react-native';
export {render};

const Stack = createStackNavigator();
export const MockedNavigator = ({component, params = {}}) => {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen
          name="MockedScreen"
          component={component}
          initialParams={params}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
};
