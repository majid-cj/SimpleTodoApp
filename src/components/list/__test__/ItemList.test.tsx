import 'react-native';
import React from 'react';
import {render} from '../../../utils/test-util';
import {ItemList} from '..';
import {TaskList, TodoList} from '../../../constants';

it('test ItemList to match snapshot', () => {
  const items: TaskList = [{id: '1', title: 'test', date: '2020-12-12'}];
  const {toJSON} = render(<ItemList isTask items={items} />);
  expect(toJSON()).toMatchSnapshot();
});

it('test ItemList Task', () => {
  const items: TaskList = [{id: '1', title: 'test', date: '2020-12-12'}];
  const {getAllByTestId} = render(<ItemList isTask items={items} />);
  const taskItems = getAllByTestId('task_item');
  expect(taskItems.length).toBe(1);
});

it('test ItemList Todo', () => {
  const items: TodoList = [
    {id: '1', task: '1', description: 'test', done: false},
    {id: '2', task: '1', description: 'test 2', done: true},
  ];
  const {getAllByTestId} = render(<ItemList items={items} />);
  const taskItems = getAllByTestId('todo_item');
  expect(taskItems.length).toBe(2);
});
