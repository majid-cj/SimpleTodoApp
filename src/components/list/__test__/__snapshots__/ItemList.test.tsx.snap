// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ItemList to match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingVertical": 8,
    }
  }
>
  <RCTScrollView
    ListEmptyComponent={
      <EmptyList
        error="list is empty !"
      />
    }
    data={
      Array [
        Object {
          "date": "2020-12-12",
          "id": "1",
          "title": "test",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    testID="item_list"
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#00416a",
              "borderRadius": 8,
              "height": 56,
              "margin": 4,
              "opacity": 1,
              "padding": 8,
            }
          }
          testID="task_item"
        >
          <Text
            style={
              Object {
                "color": "#e4e5e6",
                "fontSize": 18,
                "fontWeight": "500",
                "marginVertical": 2,
                "textAlign": "left",
                "textAlignVertical": "center",
              }
            }
          >
            test
          </Text>
          <Text
            style={
              Object {
                "color": "#f64f59",
                "fontSize": 12,
                "fontWeight": "500",
                "marginVertical": 2,
                "textAlign": "left",
                "textAlignVertical": "center",
              }
            }
          >
            2020-12-12
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
